src/models/CourseBill.js

```diff
@@ -9,7 +10,14 @@ const CourseBillSchema = mongoose.Schema({
     description: { type: String },
   },
   company: { type: mongoose.Schema.Types.ObjectId, ref: 'Company', required: true },
-  courseFundingOrganisation: { type: mongoose.Schema.Types.ObjectId, ref: 'CourseFundingOrganisation' },
+  payer: {
+    type: mongoose.Schema({
+      fundingOrganisation: { type: mongoose.Schema.Types.ObjectId, ref: 'CourseFundingOrganisation' },
+      company: { type: mongoose.Schema.Types.ObjectId, ref: 'Company' },
+    }, { _id: false, id: false }),
+    required: true,
+    validate() { return !(!!this.payer.company === !!this.payer.fundingOrganisation); },
```

Comment:
Je ne suis pas sure de comprendre pourquoi tu as ajouté cette validation. Est-ce que c'est pour etre sur qu'il y ait soit une company soit une fundingOrganisation, mais pas les 2 ?

Reply:
oui c'est ça. si il y a ni l'un ni l'autre on a !(false === false) ce qui renvoie false, si on a les deux ça donne !(true === true) ce qui renvoie false et si on a que l'un ou l'autre ça renvoie !(true === false) ce qui renvoie true. Mais du coup je me dis que je me complique un peu la tâche et que je peux mettre directement !!this.payer.company !== !!this.payer.fundingOrganisation